### PR TITLE
DX-1264: Unify exports

### PR DESCRIPTION
**Pull request by ytkimirti**
This PR introduces some small tweaks to unify some parts of our javascript packages. Here are the changes:

- Used "latest" version in examples for this package
- Updated tsup config
- Updated exports field in package.json

>NOTE: This should not bring any breaking changes with the old version. Here is the old and new output coming from the attw cli:

r before

r after

- `node.js` and `node.mjs` exports are removed since they did not resolve to a file and do not work in the prod version.
- `cloudflare.js`, `cloudflare.mjs`, `fastly.js` and `fastly.mjs` exports are kept for backwards compatibility.

### CODE REVIEW

**Comment by CahidArda** [comment]
tests are not passing @ytkimirti

File: package.json [hunk 1]
```diff
@@ -61,19 +79,6 @@
     "url": "https://github.com/upstash/upstash-redis/issues"
   },
   "homepage": "https://github.com/upstash/upstash-redis#readme",
-  "typesVersions": {
```
Comment (fahreddinozcan) [inline package.json:64]: Do we have any differences between type exports? Maybe we should add the type exports to `exports` map, now that we've removed it from here.

File: examples/cloudflare-workers/package.json [hunk 2]
```diff
@@ -12,6 +12,6 @@
     "wrangler": "^2.20.0"
   },
   "dependencies": {
-    "@upstash/redis": "latest"
```
Comment (fahreddinozcan) [inline examples/cloudflare-workers/package.json:15]: Why not `latest` here?

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -44,14 +44,8 @@ jobs:
 
       - name: Publish
         if: "!github.event.release.prerelease"
-        working-directory: ./dist
-        run: |
-          npm pkg delete scripts.prepare
```
Comment (fahreddinozcan) [inline .github/workflows/release.yml:49]: Did we remove this line, since npm doesn't already include it in release?

Comment (ytkimirti) [reply .github/workflows/release.yml:49]: It includes it but does not run the script when doing install. But now I learned that it runs the script for git installs etc. I am adding this part back

File: package.json (hunk 1 above)
Comment (ytkimirti) [reply package.json:64]: For cf and fastly exports here, typescript finds the appropriate .d.ts file itself, we don't have to specify it.

And for the `nodejs` here, I checked and it does not work in the prod version so we don't need it here

**Review by CahidArda — DISMISSED** [review]

**Review by CahidArda — APPROVED** [review]

**Review by CahidArda — APPROVED** [review]

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -54,4 +54,4 @@ jobs:
         working-directory: ./dist
         run: |
           npm pkg delete scripts.prepare
-          npm publish --access public --tag=next
```
Comment (fahreddinozcan) [inline .github/workflows/release.yml:57]: Why do we release a canary after we release a normal version

Comment (ytkimirti) [reply .github/workflows/release.yml:57]: This only works for prereleases because of the if
`if: "github.event.release.prerelease"`

Comment (fahreddinozcan) [reply .github/workflows/release.yml:57]: Ah didn't saw the `!` difference there, right

**Review by fahreddinozcan — APPROVED** [review]

**Pull request merged**